packages/common/container-definitions/src/runtime.ts

```diff
@@ -102,6 +102,17 @@ export interface IRuntime extends IDisposable {
      * @param snapshot - snapshot created at attach time
      */
     notifyAttaching(snapshot: ISnapshotTreeWithBlobContents): void;
+
+    /**
+     * Exposes the entryPoint for the container runtime.
+     * Use this as the primary way of getting access to the user-defined logic within the container runtime.
+     *
+     * @see {@link IContainer.getEntryPoint}
+     *
+     * @remarks The plan is that eventually IRuntime will no longer have a request() method, this property will no
+     * longer be optional, and it will become the only way to access the entryPoint for the runtime.
+     */
+    readonly entryPoint?: Promise<FluidObject | undefined>;
```

Comment:
I've been thinking about this a bit, and i think a method, `getEntryPoint()` might be better property in all the usages. Primarily because i don't think a Promise property is very ergonomic. There are really two ways to use it 
```
const entryPoint: FluidObject<IThing> | undefined = await runtime.entryPoint;
const thing = entryPoint?.Ithing;
```
or
```
const thing = (await runtime.entryPoint as  FluidObject<IThing> | undefined )?,Ithing
```
the second pattern should be avoided, as it uses an unsafe cast, but i think the property pattern encourages that usage. I think the first pattern becomes more natural/obvious with a method

```
const entryPoint: FluidObject<IThing> | undefined = await runtime.getEntryPoint?.();
const thing = entryPoint?.Ithing;
```

The method pattern also makes it a bit more obvious that it could fail and throw an exception

Reply:
Just as a data point, I'm not sure that the property/method distinction would make the second pattern less natural for me. But happy to make it a method everywhere since in the upper layers it does have other advantages, and I do agree a method suggests "this might throw" more than a property does.

I'll give some thought to if it'd be better to change the dataStore-level entryPoint to also be a method (in another PR, if so).